src/utils/object.js

```diff
@@ -1,10 +1,14 @@
 export function updateExistingProps(obj, props) {
   let changed = false;
 
-  for (const [key, value] of Object.entries(props)) {
-    if (obj[key] !== undefined) {
-      obj[key] = value;
-      changed = true;
+  for (let keys = Object.keys(props), len = keys.length, i = 0; i < len; ++i) {
```

Comment:
Figured there's no reason to allocate the Object.entries object.  

Historically for...of loops and de-structuring have been less performant than a traditional for loop. Not sure if that's still true, but old habits...